backend/src/main/java/org/eclipse/tractusx/puris/backend/demand/logic/services/DemandService.java

```diff
@@ -123,12 +123,12 @@ protected List<String> validateReportedDemand(Demand demand) {
         List<String> errors = new ArrayList<>();
         Partner ownPartnerEntity = partnerService.getOwnPartnerEntity();
         if (!mprService.partnerOrdersProduct(demand.getMaterial(), demand.getPartner())) {
-            errors.add("Cannot order specified material from Partner.");
+            errors.add(String.format("Partner '%s' is not configured to buy your material '%s'.", demand.getPartner().getBpnl(), demand.getMaterial().getOwnMaterialNumber()));
         }
         if ((demand.getSupplierLocationBpns() != null  && 
             ownPartnerEntity.getSites().stream().noneMatch(site -> site.getBpns().equals(demand.getSupplierLocationBpns())))
             || demand.getPartner().getSites().stream().noneMatch(site -> site.getBpns().equals(demand.getDemandLocationBpns()))) {
-            errors.add("Invalid demand: supplier or demand location is not valid.");
+            errors.add("Supplier or demand location is not valid.");
```

Comment:
```suggestion
            errors.add(String.format("Either supplier location '%s' is not configured as your site or demand location '%s' is not configured as site for customer partner '%s'.", demand.getSupplierLocationBpns(), demand.getDemandLocaitonBpns(), demand.getPartner().getBpnl()));
```
Sorry again, seem to have missed to specify the details here.